src/lib.rs

```diff
@@ -0,0 +1,704 @@
+				let mut channel_monitor_mut_references = Vec::new();
+				for (_, channel_monitor) in channel_monitors.iter_mut() {
+					channel_monitor_mut_references.push(channel_monitor);
+				}
+				let read_args = ChannelManagerReadArgs::new(
+					Arc::clone(&keys_manager),
+					Arc::clone(&chain_access),
+					Arc::clone(&chain_monitor),
+					Arc::clone(&chain_access),
+					Arc::clone(&logger),
+					user_config,
+					channel_monitor_mut_references,
+				);
+				<(BlockHash, ChannelManager)>::read(&mut f, read_args)?
+			} else {
+				// We're starting a fresh node.
+				let latest_block_height = chain_access.get_height()?;
+				let latest_block_hash = chain_access.get_block_hash(latest_block_height as u64)?;
+
+				let chain_params = ChainParameters {
+					network: config.network,
+					best_block: BestBlock::new(latest_block_hash, latest_block_height),
+				};
+				let fresh_channel_manager = channelmanager::ChannelManager::new(
+					Arc::clone(&chain_access),
+					Arc::clone(&chain_monitor),
+					Arc::clone(&chain_access),
+					Arc::clone(&logger),
+					Arc::clone(&keys_manager),
+					user_config,
+					chain_params,
+				);
+				(latest_block_hash, fresh_channel_manager)
+			}
+		};
+
+		let channel_manager = Arc::new(channel_manager);
+
+		// Step 8: Give ChannelMonitors to ChainMonitor
+		for (_blockhash, channel_monitor) in channel_monitors.drain(..) {
+			let funding_outpoint = channel_monitor.get_funding_txo().0;
+			chain_monitor.watch_channel(funding_outpoint, channel_monitor).unwrap();
+		}
+
+		// TODO: Use RGS on first start, if configured
+		// Step 10: Initialize the P2PGossipSync
+		let network_graph =
+			Arc::new(util::read_network_graph(Arc::clone(&config), Arc::clone(&logger))?);
+		let gossip_sync = Arc::new(P2PGossipSync::new(
+			Arc::clone(&network_graph),
+			None::<Arc<dyn Access + Send + Sync>>,
+			Arc::clone(&logger),
+		));
+
+		//// Step 11: Initialize the PeerManager
+		let mut ephemeral_bytes = [0; 32];
+		rand::thread_rng().fill_bytes(&mut ephemeral_bytes);
+		let lightning_msg_handler = MessageHandler {
+			chan_handler: Arc::clone(&channel_manager),
+			route_handler: Arc::clone(&gossip_sync),
+		};
+
+		let peer_manager: Arc<PeerManager> = Arc::new(PeerManager::new(
+			lightning_msg_handler,
+			keys_manager.get_node_secret(Recipient::Node).unwrap(),
+			&ephemeral_bytes,
+			Arc::clone(&logger),
+			Arc::new(IgnoringMessageHandler {}),
+		));
+
+		// Step 12: Initialize routing ProbabilisticScorer
+		let scorer = Arc::new(Mutex::new(util::read_scorer(
+			Arc::clone(&config),
+			Arc::clone(&network_graph),
+			Arc::clone(&logger),
+		)));
+
+		// Step 13: Init payment info storage
+		// TODO: persist payment info to disk
+		let _inbound_payments = Arc::new(Mutex::new(HashMap::new()));
+		let _outbound_payments = Arc::new(Mutex::new(HashMap::new()));
+
+		// Step 14: Handle LDK Events
+		let event_queue = mpsc::sync_channel(CHANNEL_BUF_SIZE);
```

Comment:
Ah, wait, but that requirement is for LDK events, right? The `event_queue` here is meant for `LdkLiteEvents`.

The idea would be: LDK events get handled by `event::LdkLiteEventHandler`, which processes them and emits more user-friendly `LdkLiteEvents` (possibly wrapping the original events) to said event queue. Yes, these events should probably not get lost either, but it's maybe not as critical if we already handled technicalities for, e.g, `FundingGenerationReady` or `SpendableOutputs`  in the event handler?

Reply:
Yes, I payment history will be persisted directly by the event handler I assume. 

But, I'm honestly not sure how we could ensure that no data is lost ever when the user doesn't process the events given. If event queue gets dropped, the user will probably not notified again about the queued events. However, I'm not sure if "we hand you individual events and it's on you if you don't act on them before dropping" is that different from "we hand you a list/queue of events, and it's on you if you don't act on all of them before dropping" at that point?

Reply:
Hm, but I'm not sure exposing this kind of interface is in line with the usability we're trying to achieve with `LdkLite`.  I agree we should handle all security-critical things before we even notify the user, but forcing him to handle events inline is sometimes just not practical? Also, it would likely force us to block the event handler and thereby the entire BP thread? Also not sure the given example matches exactly, because IIUC at least under iOS the app would be suspended and its state saved until it is brought to the foreground again? (Which opens a whole other topic on whether we would cleanly recover and continue operation from different states, but that's 'future work').

Reply:
regarding event reliability, as long as these events represent non-critical notifications type of things, it would have been fine and maybe we can loose those events.
But the problem is do we expect clients to poll through this ? I don't think that is ideal. 

Hence handling individual events using event handler is definitely better UX. 
Also, it might be helpful to have some distinction b/w core/critical events and some notification/alarm kind of things.